avaframe/ana5Utils/distanceTimeAnalysis.py

```diff
@@ -0,0 +1,792 @@
+
+    # extract front and average values of result parameter
+    if cfg['VISUALISATION'].getboolean('TTdiagram'):
+        mtiInfo = extractFrontAndMeanValues(cfgRangeTime, fields[rangeTimeResType], demHeader, mtiInfo)
+    else:
+        # extract front in simulation result for each time step
+        mtiInfo = extractFrontInSim(fields[rangeTimeResType], mtiInfo,
+             cfgRangeTime['GENERAL'].getfloat('thresholdResult'))
+        # extract average values along range gates for coloring range-time plot
+        mtiInfo = extractMeanValuesAtRangeGates(cfgRangeTime, fields[rangeTimeResType], mtiInfo)
+
+    # append time step info
+    mtiInfo['timeList'].append(t)
+
+    # remove saving time steps that have already been saved
+    dtRangeTime = com1DFA.updateSavingTimeStep(dtRangeTime, cfgRangeTime['GENERAL'], t)
+
+    return mtiInfo, dtRangeTime
+
+
+def fetchTimeStepFromName(pathNames):
+    """ split path name to fetch info on time step
+
+        Parameters
+        -----------
+        pathName: pathlib path or list of paths
+            path to file including time step info as _t%.2f_
+
+        Returns
+        -------
+        timeStep: list
+            actual time steps
+        indexTime: numpy array
+            index of timeStep list to order them in ascending order
+    """
+
+    if isinstance(pathNames, list) is False:
+        pathNames = [pathNames]
+
+    timeSteps = []
+    for fName in pathNames:
+        timeSteps.append(float(fName.stem.split('_t')[1]))
+
+    indexTime = np.argsort(np.array(timeSteps))
+
+    return timeSteps, indexTime
+
+
+def approachVelocity(mtiInfo):
+    """ compute approach velocity based on front location and time step
+
+        Parameters
+        -----------
+        mtiInfo: dict
+            info on distance to front and time steps
+
+        Returns
+        --------
+        maxVel: float
+            max value of approach velocity
+    """
+
+    # load lists
+    timeList = mtiInfo['timeList']
+    rangeList = mtiInfo['rangeList']
+
+    # as these might be not in ascending order timestep wise - order first
+    timeListSorted = np.asarray(timeList)[np.argsort(timeList)]
+    # convert to only positive distance values (move reference point to end of ava)
+    # to get correct distance increments for velocity computation
+    # nanmin required as if FV is zero range is nan
+    rangeListSorted = np.asarray(rangeList)[np.argsort(timeList)] + abs(np.nanmin(np.asarray(rangeList)))
+    approachVelocity = []
+    maxVel = 0.0
+    rangeVel = 0.0
+    # compute deltaDistance/deltaTime to get approach velocity
+    # use interval of 2*dtSave for little smmoothing
+    for index, range in enumerate(rangeListSorted[1:-1]):
+        vel = (rangeListSorted[index+1] - rangeListSorted[index-1]) / (timeListSorted[index+1] - timeListSorted[index-1])
+        approachVelocity.append(np.abs(vel))
```

Comment:
why not also return the approachVelocity? if not returning it why do you store it?

Reply:
removed it